### PR TITLE
Implement ADR015

### LINKED ISSUE #1215: [FEATURE] Implement ADR015
https://apalache.informal.systems/docs/adr/015adr-trace.html

### PR DESCRIPTION
**Pull request by konnov**
This PR closes #1215. It implements one more counterexample output. This time in the [Informal Trace Format](https://apalache.informal.systems/docs/adr/015adr-trace.html).

It may happen that the integration tests will fail, as my local build is blocked by #1219.

- [x] Tests added for any new code
- [x] Ran `make fmt-fix` (or had formatting run automatically on all files edited)
- [ ] ~Documentation added for any new functionality~
- [x] Entry added to [UNRELEASED.md](./UNRELEASED.md) for any new functionality

### CODE REVIEW

File: tla-io/src/main/scala/at/forsyte/apalache/io/lir/ItfCounterexampleWriter.scala [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+   * @return the JSON representation of the counterexample in the ITF format
+   */
+  def mkJson(rootModule: TlaModule, states: List[NextState]): ujson.Value = {
+    // merge constant initialization and variable initialization into a single state
+    val state0 = states match {
+      case Seq(constInit)                => constInit._2
+      case Seq(constInit, initState, _*) => constInit._2 ++ initState._2
+      case Seq()                         => throw new IllegalArgumentException("Expected at least one state, found none")
+    }
+    val mappedStates = state0 :: states.drop(2).map(_._2)
```
Comment (Kukovec) [inline tla-io/src/main/scala/at/forsyte/apalache/io/lir/ItfCounterexampleWriter.scala:44]: If `states: List[NextState]`, you probably shouldn't be matching `Seq(...)`, but 
```suggestion
  def mkJson(rootModule: TlaModule, states: List[NextState]): ujson.Value = {
    // merge constant initialization and variable initialization into a single state
    val state0 = states match {
      case constInit :: Nil                => constInit._2
      case constInit :: initState :: _ => constInit._2 ++ initState._2
      case Nil                     => throw new IllegalArgumentException("Expected at least one state, found none")
    }
    val mappedStates = state0 :: states.drop(2).map(_._2)
```
You could also just do:
```suggestion
  def mkJson(rootModule: TlaModule, states: List[NextState]): ujson.Value = {
    // merge constant initialization and variable initialization into a single state
    if (states.isEmpty) throw new IllegalArgumentException("Expected at least one state, found none")
    val (prefix, suffix) = states.splitAt(2)
    val mappedStates = prefix.reduce( _ ++ _ ) :: suffix.map(_._2)
```

File: tla-io/src/main/scala/at/forsyte/apalache/io/lir/ItfCounterexampleWriter.scala [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+      ujson.Str(str)
+
+    case ex @ OperEx(TlaFunOper.tuple, args @ _*) =>
+      ex.typeTag match {
+        case Typed(SeqT1(_)) =>
+          ujson.Arr(args.map(exToJson): _*)
+
+        case _ =>
+          ujson.Obj("#tup" -> ujson.Arr(args.map(exToJson): _*))
+      }
```
Comment (Kukovec) [inline tla-io/src/main/scala/at/forsyte/apalache/io/lir/ItfCounterexampleWriter.scala:104]:
```suggestion
      val arr = ujson.Arr(args.map(exToJson)
      ex.typeTag match {
        case Typed(SeqT1(_)) =>
          ujson.Arr(arr: _*)

        case _ =>
          ujson.Obj("#tup" -> arr: _*))
      }
```

**Review by Kukovec — APPROVED** [review]
LGTM, minor nits.

**Review by gabrielamafra — APPROVED** [review]

File: UNRELEASED.md [hunk 3]
```diff
@@ -10,6 +10,10 @@
          * Some bug fix, see #124
 
      DO NOT LEAVE A BLANK LINE BELOW THIS PREAMBLE -->
+### Features
+
+ * Output in the Informal Trace Format, see #1215
```
[the remark this answers is not in the document]
Comment (Kukovec) [reply UNRELEASED.md:15]: Typically, the changelog references the issue being resolved, which is https://github.com/informalsystems/apalache/issues/1215. In this case, it's a dummy issue, but still.

[the remark this answers is not in the document]
Comment (gabrielamafra) [reply UNRELEASED.md:15]: Ok, I just saw that bot mine and Shon's entry refer to the PR. I'll be careful to use the issue from now on.

[the remark this answers is not in the document]
Comment (shonfeder) [reply UNRELEASED.md:15]: Yeah, I generally refer to them PR. I didn’t know we were meant to link an issue either :)

File: tla-io/src/main/scala/at/forsyte/apalache/io/lir/ItfCounterexampleWriter.scala (hunk 2 above)
Comment (konnov) [reply tla-io/src/main/scala/at/forsyte/apalache/io/lir/ItfCounterexampleWriter.scala:104]: Good point. I did not notice that it was a list :)

File: UNRELEASED.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (Kukovec) [reply UNRELEASED.md]: Huh, I guess this is a style thing. For me, the issue is usually the one that has a description of what the problem was, so linking it tells you what was resolved.

[the remark this answers is not in the document]
Comment (konnov) [reply UNRELEASED.md]: The issue and the PR are cross-linked. In this case, the PR is more informative, so let's refer to the PR.

**Pull request merged**